.github/2.0/cookbooks/Flow.md

```diff
@@ -431,6 +431,92 @@ f = (Flow()
 
 </table>
 
+#### Commonly used arguments for distribution setup in `.add`
+
+| Name | default | Description |
+| --- | --- | --- |
+| `host` | `0.0.0.0` | The host of the machine. Can be an ip address or DNS name (e.g. `0.0.0.0`, `my_encoder.jina.ai`) |
+| `port_expose` | randomly initialized | Port of JinaD on the remote machine. |
+| `port_in` | randomly initialized | Port for incoming traffic for the Executor. |
+| `port_out` | randomly initialized | Port for outgoing traffic for the Executor. This is only used in the remote-local use-case described below. |
+| `connect_to_predecessor` | `False` | Forces a Head to connect to the previous Tail. This is only used in the remote-local use-case described below. |
+| `external` | `False` | Stops `Flow` from context managing an Executor. This allows spawning of an external Executor and reusing across multiple Flows. |
+| `uses`, `uses_before` and `uses_after` prefix | No prefix | When prefixing one of the `uses` arguments with `docker` or `jinahub+docker`, the Pod does not run natively, but is spawned inside a container. |
+
+#### Commonly used patterns for `.add`
+
+##### Local native Executor via `.py` file:
+
+```python
+class MyExecutor():...
+
+f.add(uses=MyExecutor)
+f.add(uses='MyExecutor')
+```
+
+##### Local native Executor via `.yml` file:
+
+```python
+f.add(uses='mwu_encoder.yml')
+```
+
+##### Local docker Executor:
+
+```python
+f.add(uses='docker://MyExecutor')
+```
+
+##### Local docker Executor from jinahub:
+
+```python
+f.add(uses='jinahub+docker://MyExecutor')
+```
+
+##### Using an already spawned Executor:
+
+The external Executor in the following two use-cases could have been spawned
+
+- either by another Flow
+- or by the `jina executor` CLI command
+
+```python
+f.add(host='localhost', port_in=12345, external=True)
+f.add(host='123.45.67.89', port_in=12345, external=True)
+```
+
+##### Remote Executor
+
+In any of the following remote patterns without `external`, JinaD must be running on the remote machine.
+Furthermore, JinaD must be setup to listen on `port_expose`.
+
+```python
+f.add(uses='mwu_encoder.yml', host='123.45.67.89', port_in=12345, port_expose=8080)
+```
+
+`uses` can take any argument as in the local case (e.g. `jinahub+docker:://...`).
+
+##### Setting the Gateway data port and host
+
+Whenever the networking of the Gateway should be changed, the respective arguments go directly into the Flow initialization.
```

Comment:
~Whenever the networking of the Gateway should be changed~ If you need to change the networking of the Gateway